### PR TITLE
[SPARK-35378][SQL][FOLLOWUP] isLocal should consider CommandResult

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
#32513 added the case class `CommandResult` so as we can eagerly execute command locally. But we forgot to update 
`isLocal` of `Dataset`.

### Why are the changes needed?
`Dataset.isLocal` should consider `CommandResult`.

### Does this PR introduce _any_ user-facing change?
Yes. If the SQL plan is `CommandResult`, `Dataset.isLocal` must return true.

### How was this patch tested?
No test.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 1]
```diff
@@ -597,7 +597,8 @@ class Dataset[T] private[sql](
    * @group basic
    * @since 1.6.0
    */
-  def isLocal: Boolean = logicalPlan.isInstanceOf[LocalRelation]
+  def isLocal: Boolean = logicalPlan.isInstanceOf[LocalRelation] ||
+    logicalPlan.isInstanceOf[CommandResult]
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:601]: Could you add a test coverage for this, @beliefer  ?

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:601]: +1, this API has no test as all, we can add one in `DataFrameSuite`

**Comment by dongjoon-hyun** [comment]
Gentle ping, @beliefer .

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala (hunk 1 above)
Comment (beliefer) [reply sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala:601]: Thank you for reminder.

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala [hunk 2]
```diff
@@ -2907,6 +2907,15 @@ class DataFrameSuite extends QueryTest
       }
     }
   }
+
+  test("isLocal should consider CommandResult and LocalRelation") {
+    withTable("t1") {
+      val df = sql("CREATE TABLE t USING PARQUET AS SELECT 1 as a")
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala:2913]: can we use `SHOW TABLES` which returns data? Then we don't need to create a table and wrap with `withTable`.

Comment (beliefer) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala:2913]: OK

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @beliefer and @cloud-fan .

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by beliefer** [comment]
@dongjoon-hyun @cloud-fan Thanks for review.

**Pull request closed without merging**